### PR TITLE
Refactored example to demo. Removed extraneous example files.

### PR DESCRIPTION
**Pull request by abradle**

### CODE REVIEW

**Comment by abradle** [comment]
I've just updated the pom to work with a release of mmtf-spark in Maven central. (0.0.1). 

This will  at least mean that the builds will work going forward too. (This first one might not work as I literally just released to maven central).

File: src/main/java/demo/ClusterCalphas.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	/**
+	 * 
+	 * @param args
+	 * @throws IOException
+	 */
+	public static void main(String[] args) throws IOException {
+		// Get the underlying C-alpha chains
+//		List<Tuple2<String, Segment>> calphaChains = SparkUtils.getCalphaChains(new String[] {"4CUP","4CUQ","4CUT","4CUR","4CUS","1STP"}).filterMinLength(10).getSegmentRDD().collect();
+		List<Tuple2<String, Segment>> listChains = new StructureDataRDD("/Users/anthony/full").sample(0.01).getCalpha().getSegmentRDD().collect();
```
Comment (andreasprlic) [inline src/main/java/demo/ClusterCalphas.java]: Can this List be made a RDD? Should make it more scaleable

File: src/main/java/demo/TmScore.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * @author Anthony Bradley
+ *
+ */
+public class TmScore implements Function<Tuple2<Integer,Integer>,Tuple3<Long, Long, Double>> {
+
+	/**
+	 * The serial version of this commit.
+	 */
+	private static final long serialVersionUID = -2557285958500330806L;
+	private Broadcast<List<Tuple2<String, Segment>>> calphaChains;
```
Comment (andreasprlic) [inline src/main/java/demo/TmScore.java]: This means that every node needs to keep a copy of all data in memory, right? I guess this should work due to mmtf, but it also feels like there should be a more efficient way of doing this, which does not require a List

File: src/main/java/org/biojava/spark/function/TmScorer.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		}  
+		return tmScore;
+	}
+
+	/**
+	 * Gets the C-alpha {@link Atom} for the given input {@link Segment}.
+	 * @param segment the input {@link Segment} object
+	 * @return the C-alpha array of {@link Atom} objects
+	 */
+	private static Atom[] getCaAtoms(Segment segment) {
```
Comment (andreasprlic) [inline src/main/java/org/biojava/spark/function/TmScorer.java]: We should take a look tomorrow if Segment could implement the StructureIdentifier interface.

File: src/main/java/demo/ClusterCalphas.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		int numChains = listChains.size();
+		Broadcast<List<Tuple2<String, Segment>>> calphaChains = SparkUtils.getSparkContext().broadcast(listChains);
```
Comment (andreasprlic) [inline src/main/java/demo/ClusterCalphas.java]: Again, feels wrong to have to reduce two lines earlier, just so you can broadcast this here.

File: src/main/java/demo/ClusterCalphas.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		System.out.println("Analysisng " + numChains + " chains");
+		// Now apply a function to those chains 
+		JavaPairRDD<Integer, Integer> comparisons = SparkUtils.getComparisonMatrix(numChains);
```
Comment (andreasprlic) [inline src/main/java/demo/ClusterCalphas.java]: I think this should be a `Tuple2<Tuple2<String,Segment>,Tuple2<String,Segment>>` here, so you don't need the Broadcast list from above.

File: src/main/java/demo/ClusterCalphas.java [hunk 6]
```diff
@@ -26,20 +29,24 @@
...
+		int numChains = listChains.size();
+		Broadcast<List<Tuple2<String, Segment>>> calphaChains = SparkUtils.getSparkContext().broadcast(listChains);
+		System.out.println("Analysisng " + numChains + " chains");
 		// Now apply a function to those chains 
-		JavaRDD<Tuple3<Long, Long, Double>> similarities = SparkUtils.getComparisonMatrix(numChains).map(new TmScore(calphaChains));
+		JavaPairRDD<Integer, Integer> comparisons = SparkUtils.getComparisonMatrix(numChains);
+		System.out.println("Performing "+comparisons.count()+" comparisons");
+		JavaRDD<Tuple3<Long, Long, Double>> similarities = comparisons.map(new TmScore(calphaChains));
+		// Perform the clusterings
 		PowerIterationClustering pic = new PowerIterationClustering()
```
Comment (andreasprlic) [inline src/main/java/demo/ClusterCalphas.java]: about clustering: just found this:

https://issues.apache.org/jira/browse/SPARK-2429

Comment (andreasprlic) [inline src/main/java/demo/ClusterCalphas.java]: and while we are at it, also v. interesting:

https://github.com/derrickburns/generalized-kmeans-clustering

**Comment by abradle** [comment]
Build should work now mmtf-spark 0.0.1 is in maven central. 

@andreasprlic good points - will discuss after your meeting if you have time.

**Comment by abradle** [comment]
So build is still failing - because it needs the new Chain model. So in a few days - when that pull request is accepted - it will work.

**Pull request closed without merging**